packages/web3torrent/src/utils/copy-to-clipboard.ts

```diff
@@ -0,0 +1,28 @@
+export const clipboardCopy = (text: string) => {
+  const span = document.createElement('span');
+  span.textContent = text;
+
+  document.body.appendChild(span);
+
+  const selection = window.getSelection();
+  const range = window.document.createRange();
+  if (selection) {
+    selection.removeAllRanges();
+    range.selectNode(span);
+    selection.addRange(range);
+  }
+  let success = false;
+  try {
+    success = window.document.execCommand('copy');
+  } catch (err) {
+    console.error('error', err);
+  }
+  if (selection) {
+    selection.removeAllRanges();
+  }
+  window.document.body.removeChild(span);
+
+  return success
+    ? Promise.resolve()
+    : Promise.reject(new DOMException('The request is not allowed', 'NotAllowedError'));
```

Comment:
What does `The request is not allowed` mean in this context? Can we use a better error message? AFAICT the concept of a request can be a bit weird to bring up during a copy/paste operation.

Reply:
ummm okay :-)